### PR TITLE
feat(sink): support sink rate limit for external sink

### PR DESCRIPTION
**Pull request by hzxa21**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Closes https://github.com/risingwavelabs/risingwave/issues/16253

Support `SET SINK_RATE_LIMIT` and `ALTER SINK ... SET SINK_RATE_LIMIT` for external sink. Sink into table is not supported.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

Support session config `SET SINK_RATE_LIMIT to ...` and `ALTER SINK ... SET SINK_RATE_LIMIT to ...` for external sink. Sink into table is not supported. See example in [rate_limit.slt](https://github.com/risingwavelabs/risingwave/blob/aae18eaf3673cfe00172dbfcd0dc7a798c454e05/e2e_test/sink/rate_limit.slt#L10)

</details>

### CODE REVIEW

**Review by lmatz — APPROVED** [review]

**Comment by tabVersion** [comment]
quick question: if the sink rate limit makes more data buffered in LogStore?

**Comment by hzxa21** [comment]
> quick question: if the sink rate limit makes more data buffered in LogStore?

If sink decoupled is on, yes. If sink decoupled is off, it will cause backpressure.

**Review by MrCroxx — APPROVED** [review]
LGTM

**Comment by MrCroxx** [comment]
These logs are keep repeating. Let me try to fix.

```
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
2022-10-11T06:02:13.840593Z  WARN actor{otel.name="Actor 231" actor_id=231 prev_epoch=3160993419558912 curr_epoch=3160993541062656}:executor{otel.name="Sink E700000001"}: risingwave_connector::sink::log_store: rate limit control channel closed
```

File: src/connector/src/sink/log_store.rs [hunk 1]
```diff
@@ -324,6 +333,252 @@ impl<R: LogReader> LogReader for MonitoredLogReader<R> {
...
+        loop {
+            select! {
+                biased;
+                rate_limit_change = pin!(self.control_rx.recv()) => {
+                    if let Some(new_rate_limit) = rate_limit_change {
+                        let prev = self.update_rate_limit(new_rate_limit);
+                        paused = self.rate_limit == Some(0);
+                        tracing::info!("rate limit changed from {:?} to {:?}, paused = {paused}", prev, self.rate_limit);
+                    } else {
+                        tracing::warn!("rate limit control channel closed");
```
Comment (MrCroxx) [inline src/connector/src/sink/log_store.rs:529]: Should the log reader exit here? Seems there is no way to restore the rate limit control channel.

Comment (MrCroxx) [reply src/connector/src/sink/log_store.rs:529]: Simulation recovery test passed with `bail!` here.

Comment (MrCroxx) [reply src/connector/src/sink/log_store.rs:529]: Is it okay that I commit the fix directly in this PR?

**Pull request merged**